### PR TITLE
fix(server): configurable file paths for test isolation (#429)

### LINKED ISSUE #424: Add unit tests for permission-hook.js shared module
## Context

Identified during review of PR #423.

## Description

The new `permission-hook.js` module was extracted from `cli-session.js` and is now shared between CLI and PTY modes. The existing tests cover CliSession's behavior (which transitively tests the hook via integration), but there are no dedicated unit tests for:

- `withSettingsLock()` serialization behavior
- `createPermissionHookManager()` registration, retry, and destroy lifecycle
- `registerPermissionHookSync()` / `unregisterPermissionHookSync()` file I/O edge cases (missing dir, corrupt JSON, concurrent writes)

## Original Review Comment

> The shared module is the critical new code in this PR. It handles file I/O with retry logic and a promise-chain lock — both are easy to get wrong and hard to debug in production. Dedicated unit tests with a mock filesystem would catch regressions early.

## Acceptance Criteria

- [ ] Unit tests for `withSettingsLock` (serialization, error propagation)
- [ ] Unit tests for `createPermissionHookManager` (register, retry backoff, destroy cancels timers, unregister)
- [ ] Unit tests for `registerPermissionHookSync` (ENOENT creates dir, corrupt JSON throws, idempotent re-registration)
- [ ] Unit tests for `unregisterPermissionHookSync` (removes hook, cleans empty objects, handles missing file)

### LINKED ISSUE #429: Bug: permission-hook tests write to REAL ~/.claude/settings.json — causes production contamination
## Context

Identified during review of PR #426.

## Description

The `permission-hook.js` module hardcodes the settings path via `homedir()` + `~/.claude/settings.json`. This means tests that exercise `register()`, `unregister()`, and idempotency must modify the real user settings file, creating several problems:

1. May interfere with a running Claude instance
2. Could corrupt existing user configuration if a test fails mid-execution
3. Makes tests non-deterministic (they skip silently in CI where the file doesn't exist)
4. Violates test isolation principles

The fix is to refactor `permission-hook.js` to accept an optional settings path parameter (or use an environment variable override), allowing tests to use a temporary directory.

## Original Comments

> These tests directly modify the real `~/.claude/settings.json` file, which is problematic. Tests should not write to the user's actual configuration file.

> Use a temporary directory approach or mock the file system operations to ensure tests run consistently regardless of the environment setup.

## Acceptance Criteria

- [ ] `registerPermissionHookSync()` and `unregisterPermissionHookSync()` accept an optional `settingsPath` parameter
- [ ] `createPermissionHookManager()` accepts an optional `settingsPath` in its options
- [ ] Tests use a temp directory for all settings.json operations
- [ ] Tests run deterministically in CI (no silent skips)
- [ ] Existing production behavior unchanged (defaults to `~/.claude/settings.json`)

### LINKED ISSUE #430: Add edge case tests for permission-hook: corrupt JSON, ENOENT, missing file unregister
## Context

Identified during review of PR #426.

## Description

The current permission-hook tests cover the happy path (register, unregister, idempotency, lock serialization) but miss several edge cases from the original #424 acceptance criteria:

- ENOENT creates directory (partially tested through register() but not explicitly verified)
- Corrupt JSON handling (not tested)
- Missing file handling in unregister (not tested)

These require the module to accept a configurable settings path (#429) to test properly without touching real files.

## Original Comment

> The acceptance criteria from issue #424 mention testing edge cases like:
> - ENOENT creates directory (partially tested through register() but not explicitly verified)
> - Corrupt JSON handling (not tested)
> - Missing file handling in unregister (not tested)
>
> While the current tests do cover the main happy paths and some concurrent execution scenarios, several edge cases mentioned in the original requirements are missing.

## Acceptance Criteria

- [ ] Test: ENOENT triggers directory creation
- [ ] Test: Corrupt/invalid JSON in settings.json is handled gracefully
- [ ] Test: unregister() with missing settings.json doesn't throw
- [ ] Test: register() retry and error emission after 3 failures
- [ ] Depends on #429 (configurable settings path) for proper isolation

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- Add `settingsPath` parameter to permission-hook.js (register/unregister/manager)
- Add `stateFilePath` parameter to session-manager.js (constructor/serialize/restore)
- Thread `settingsPath` through cli-session.js to `createPermissionHookManager`
- Restore register/unregister/idempotency tests using temp directory isolation
- Migrate session-manager serialize/restore tests to use temp dirs via `stateFilePath`

## Changes

| File | What changed |
|------|-------------|
| `src/permission-hook.js` | Accept optional `settingsPath` in all functions, default to `~/.claude/settings.json` |
| `src/session-manager.js` | Accept optional `stateFilePath` in constructor, default to `~/.chroxy/session-state.json` |
| `src/cli-session.js` | Accept `settingsPath` option, pass through to hook manager |
| `tests/permission-hook.test.js` | Restore register/unregister/idempotency tests with `mkdtempSync` isolation |
| `tests/session-manager.test.js` | Replace hardcoded `~/.chroxy/` paths with temp dir via `stateFilePath` |

## Test Plan

- [x] All 476 server tests pass
- [x] No writes to real `~/.claude/settings.json` during tests
- [x] No writes to real `~/.chroxy/session-state.json` during tests
- [x] Production behavior unchanged (defaults used when no path provided)

Closes #429, closes #430, closes #424

### CODE REVIEW

File: packages/server/tests/session-manager.test.js [hunk 1]
```diff
@@ -1,27 +1,18 @@
 import { describe, it, beforeEach, afterEach } from 'node:test'
 import assert from 'node:assert/strict'
-import { writeFileSync, readFileSync, existsSync, unlinkSync, mkdirSync } from 'fs'
+import { writeFileSync, readFileSync, existsSync, unlinkSync, mkdirSync, mkdtempSync, rmSync } from 'fs'
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/server/tests/session-manager.test.js]: Fixed in `e8414400`

**Change:** Removed unused `unlinkSync` and `mkdirSync` imports left over from the migration to `mkdtempSync`/`rmSync`

```diff
- import { writeFileSync, readFileSync, existsSync, unlinkSync, mkdirSync, mkdtempSync, rmSync } from 'fs'
+ import { writeFileSync, readFileSync, existsSync, mkdtempSync, rmSync } from 'fs'
```
